pkg/virt-operator/creation/components/crds.go

```diff
@@ -37,6 +39,44 @@ const (
 	KUBEVIRT_PROMETHEUS_RULE_NAME = "prometheus-kubevirt-rules"
 )
 
+var (
+	VIRTUALMACHINE                   = "virtualmachines." + virtv1.VirtualMachineInstanceGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCE           = "virtualmachineinstances." + virtv1.VirtualMachineInstanceGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCEPRESET     = "virtualmachineinstancepresets." + virtv1.VirtualMachineInstancePresetGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCEREPLICASET = "virtualmachineinstancereplicasets." + virtv1.VirtualMachineInstanceReplicaSetGroupVersionKind.Group
+	VIRTUALMACHINEINSTANCEMIGRATION  = "virtualmachineinstancemigrations." + virtv1.VirtualMachineInstanceMigrationGroupVersionKind.Group
+	KUBEVIRT                         = "kubevirts." + virtv1.KubeVirtGroupVersionKind.Group
+	VIRTUALMACHINESNAPSHOT           = "virtualmachinesnapshots." + snapshotv1.SchemeGroupVersion.Group
+	VIRTUALMACHINESNAPSHOTCONTENT    = "virtualmachinesnapshotcontents." + snapshotv1.SchemeGroupVersion.Group
+	PreserveUnknownFieldsFalse       = false
+)
+
+func patchValidation(crd *extv1beta1.CustomResourceDefinition) {
+	crd.Spec.PreserveUnknownFields = &PreserveUnknownFieldsFalse
+	validation, ok := CRDsValidation[crd.Spec.Names.Singular]
+	if !ok {
+		return
+	}
+	crvalidation := extv1beta1.CustomResourceValidation{}
+	k8syaml.NewYAMLToJSONDecoder(strings.NewReader(validation)).Decode(&crvalidation)
```

Comment:
I covered this in `crds_test.go`. Do you think we need to propagate it?

Reply:
I think this still needs to be updated to return an error?

Reply:
> I covered this in crds_test.go. Do you think we need to propagate it?

yes, unfortunately we need to propagate it. I don't know why this would ever occur, but if a function returns an error, we need to process it. Otherwise we run the risk of undefined behavior by silently ignoring the error.